apps/happy-blocks/src/pricing-plans/hooks/pricing-plans.ts

```diff
@@ -0,0 +1,207 @@
+import formatCurrency from '@automattic/format-currency';
+import { useEffect, useState } from '@wordpress/element';
+import { __, sprintf } from '@wordpress/i18n';
+import config from '../config';
+import { ApiPricingPlan, PricingPlansConfiguration } from '../types.js';
+
+const DAYS_IN_MONTH = 31;
+const MONTHS_IN_YEAR = 12;
+
+/**
+ * Returns a formatted motnhly price for a plan.
+ */
+const getMonthlyPrice = ( plan: ApiPricingPlan ): string | null => {
+	return formatCurrency(
+		Math.round( ( plan.raw_price / plan.bill_period ) * DAYS_IN_MONTH ),
+		plan.currency_code,
+		{
+			stripZeros: true,
+		}
+	);
+};
+
+/**
+ * Returns a description of the savings of the yearly plan over the monthly plan.
+ */
+const getSavingsDescription = (
```

Comment:
I suppose we have functionality to compute these on the backend as well since the saving could be dynamic. Could we use that, please?